include/lbann/weights/weights.hpp

```diff
@@ -63,6 +63,13 @@ class optimizer;
  */
 class weights {
   friend class optimizer;
+private:
+  weights();
+  // -----------------------------------------------
+  // Internal method for setting the comm pointer
+  // -----------------------------------------------
+  void set_comm(lbann_comm& comm);
```

Comment:
This should probably take a `shared_ptr`.

Reply:
Lets defer this.